Replace chplenv variables in ReadFileWithComments

ronawho opened the pull request:
`ReadFileWithComments` and `runSkipIf` are responsible for reading most
testing infrastructure files in sub_test. Both can either just read in a
non-executable file or run an executable file and read in the output.
`runSkipIf` will inject the chplenv for executable and non-executable
files, but `ReadFileWithComments` was only doing so for executable
files.

For non-executable files `ReadFileWithComments` was only injecting
env vars. Update it to inject chplenv vars as well.

Related prior work:
 - https://github.com/chapel-lang/chapel/commit/407f09565165d0181ce95473416f4a74ee1e2da4
 - https://github.com/chapel-lang/chapel/commit/fe51b2f45e047ce7cbc2e586caf5548ba7616b19
 - https://github.com/chapel-lang/chapel/commit/4b1eeb4a3d05b58335af0a943cd20a7fa33484c2

See https://github.com/chapel-lang/chapel/pull/16805 for motivation

ronawho commented on util/test/sub_test.py:
```diff
@@ -261,20 +261,33 @@ def PerfDirFile(s):
+    os.environ.clear()
+    os.environ.update(orig_env)
+    return new_path
+
```
Feels a bit dirty to modify os.environ just to use expandvars. Alternative implementation might be:

```py
# Similar to os.path.expandvars but stuff chplenv into it too
def expandvars_chpl(path):
    expand_env = os.environ.copy()
    expand_env.update(get_chplenv())
    return string.Template(path).substitute(expand_env)
```

but I don't have a strong preference here.

lydia-duncan replied:
Yeah, I have a slight preference for the alternative implementation

ronawho replied:
Changed

lydia-duncan commented:
So it looks like the mason EXECENVs were working around this issue a lot, and test//patterns/cltools/fileCompile/chplCompile.execenv and the UnitTest execenv.  Modifying one of those to be just a non-executable list of environment variables would be a good way to test, and then we can save the other ones for a clean up item or something

ronawho commented:
> So it looks like the mason EXECENVs were working around this issue a lot

Oddly enough these were executable EXECENVs, so they could have been using `CHPL_HOST_BIN_SUBDIR` before anyways. I adjusted them to use `CHPL_HOST_BIN_SUBDIR` and made them non-executable

> test/patterns/cltools/fileCompile/chplCompile.execenv and the UnitTest execenv.

Updated to be non-execuable and use `CHPL_HOST_BIN_SUBDIR`

c2chapel tests also uses the bin_dir in precomps, but it looks like we don't currently inject the env into pre{comp,exec,diff} files. I'll look at that in a followup PR.
